integration_test/install_test.go

```diff
@@ -105,6 +109,115 @@ func TestKrewInstall_Manifest(t *testing.T) {
 	test.AssertExecutableInPATH("kubectl-" + validPlugin)
 }
 
+func startLocalServer(t *testing.T) (*httptest.Server, string) {
+	t.Helper()
+
+	server := httptest.NewUnstartedServer(http.FileServer(http.Dir("testdata")))
+
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		t.Fatalf("trouble starting local server")
+	}
+	server.Listener = listener
+	server.Start()
+	return server, server.URL
+}
+
+func TestKrewInstall_ManifestURL(t *testing.T) {
+	skipShort(t)
+
+	server, baseURL := startLocalServer(t)
+	defer server.Close()
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.Krew("install",
+		"--manifest-url", baseURL+manifestURI).RunOrFail()
+	test.AssertExecutableInPATH("kubectl-" + validPlugin)
+}
+
+func TestKrewInstall_ManifestURLAndArchive(t *testing.T) {
+	skipShort(t)
+
+	server, baseURL := startLocalServer(t)
+	defer server.Close()
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.Krew("install",
+		"--manifest-url", baseURL+archiveManifestURI,
+		"--archive", filepath.Join("testdata", fooPlugin+".tar.gz")).
+		RunOrFail()
+	test.AssertExecutableInPATH("kubectl-" + fooPlugin)
+}
+
+func TestKrewInstall_ManifestURLAndManifest(t *testing.T) {
+	skipShort(t)
+
+	server, baseURL := startLocalServer(t)
+	defer server.Close()
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	err := test.Krew("install",
+		"--manifest-url", baseURL+manifestURI,
+		"--manifest", filepath.Join("testdata", fooPlugin+constants.ManifestExtension)).Run()
+	if err == nil {
+		t.Errorf("expected install to fail but was successful")
+	}
+}
+
+func TestKrewInstall_EmptyManifestURL(t *testing.T) {
+	skipShort(t)
+
+	server, baseURL := startLocalServer(t)
+	defer server.Close()
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	err := test.Krew("install",
+		"--manifest-url", baseURL+emtpyManifestURI).Run()
+	if err == nil {
+		t.Errorf("expected install to fail but was successful")
+	}
+}
```

Comment:
what does this really test?
our validation method? aren't those already covered with unit tests?